antarest/login/service.py

```diff
@@ -183,7 +197,7 @@ def exists_bot(self, id: int) -> bool:
         return self.bots.exists(id)
 
     def authenticate(self, name: str, pwd: str) -> Optional[JWTUser]:
-        user = self.ldap.get_by_name(name) or self.users.get_by_name(name)
+        user = self.ldap.login(name, pwd) or self.users.get_by_name(name)
```

Comment:
this will fail to authenticate a ldap user (UserLdap has no password.check).
we could add this unit test :
```
def test_authentication_ldap_user():
    users = Mock()
    users.get_by_name.return_value = None

    ldap = Mock()
    ldap.login.return_value = UserLdap(id=10, name="ExtUser")
    ldap.get_by_name.return_value = None

    service = LoginService(
        user_repo=users,
        bot_repo=Mock(),
        group_repo=Mock(),
        role_repo=Mock(),
        ldap=ldap,
        event_bus=Mock(),
    )
    assert not service.authenticate("dupond", "pwd")
    ldap.get.assert_called_once_with(10)
```